Build error in UART driver for slstk3400a with device power management

**Describe the bug**
Build error observed in weekly CI on Silabs Series 0 devices (`slstk3400a`).

**To Reproduce**
Build any application that uses serial/console with `CONFIG_PM_DEVICE=y`

**Expected behavior**
Build succeeds

**Logs and console output**
```
INFO    - 2) tests/kernel/device/kernel.device on slstk3400a/efm32hg322f64 error (Build failure)
INFO    - 3) tests/kernel/device/kernel.device.minimallibc on slstk3400a/efm32hg322f64 error (Build failure)
INFO    - 4) tests/kernel/device/kernel.device.pm on slstk3400a/efm32hg322f64 error (Build failure)
INFO    - 5) tests/lib/cpp/cxx/cpp.main.cpp14 on slstk3400a/efm32hg322f64 error (Build failure)
INFO    - 6) tests/lib/cpp/cxx/cpp.main.cpp17 on slstk3400a/efm32hg322f64 error (Build failure)
INFO    - 7) tests/lib/cpp/cxx/cpp.main.cpp20 on slstk3400a/efm32hg322f64 error (Build failure)
INFO    - 8) tests/lib/cpp/cxx/cpp.main.cpp2A on slstk3400a/efm32hg322f64 error (Build failure)
INFO    - 9) tests/lib/cpp/cxx/cpp.main.cpp2B on slstk3400a/efm32hg322f64 error (Build failure)
INFO    - 10) tests/lib/cpp/cxx/cpp.main.cpp98 on slstk3400a/efm32hg322f64 error (Build failure)

zephyr/drivers/serial/uart_gecko.c:500:58: error: 'USART_STATUS_TXIDLE' undeclared (first use in this function); did you mean 'USART_STATUS_TXBL'?
  500 |                 while (!(USART_StatusGet(config->base) & USART_STATUS_TXIDLE)) {
      |                                                          ^~~~~~~
      |                                                          USART_STATUS_TXBL
/home/mmahadevan/zephyrproject/zephyr/drivers/serial/uart_gecko.c:500:58: note: each undeclared identifier is reported only once for each function it appears in
```

--- a/drivers/serial/uart_gecko.c
+++ b/drivers/serial/uart_gecko.c
@@ -492,13 +492,15 @@ static int uart_gecko_init(const struct device *dev)
 #ifdef CONFIG_PM_DEVICE
 static int uart_gecko_pm_action(const struct device *dev, enum pm_device_action action)
 {
-	const struct uart_gecko_config *config = dev->config;
+	__maybe_unused const struct uart_gecko_config *config = dev->config;
 
 	switch (action) {
 	case PM_DEVICE_ACTION_SUSPEND:
+#ifdef USART_STATUS_TXIDLE
 		/* Wait for TX FIFO to flush before suspending */
 		while (!(USART_StatusGet(config->base) & USART_STATUS_TXIDLE)) {
 		}
+#endif
 		break;
 
 	case PM_DEVICE_ACTION_RESUME:

